frontend/src/utilities/useFetchState.ts

```diff
@@ -223,12 +223,16 @@ const useFetchState = <Type, Default extends Type = Type>(
     };
   }, [call, refreshRate]);
 
+  // Use a reference for `call` to ensure a stable reference to `refresh` is always returned
+  const callRef = React.useRef(call);
+  callRef.current = call;
```

Comment:
Interesting 🤔 I guess you probably don't want to react to and then invoke a refresh accidentally if the `call` gets rebuilt for any reason. Some hook with a standard dependency on `refresh` could have a bad day reinvoking a fetch. Nice catch 👍